wfe2/wfe.go

```diff
@@ -490,6 +490,13 @@ func (wfe *WebFrontEndImpl) NewAccount(
 
 	existingAcct, err := wfe.SA.GetRegistrationByKey(ctx, key)
 	if err == nil {
+		if existingAcct.Status == core.StatusDeactivated {
+			// If there is an existing, but deactivated account, then return an unauthorized
+			// problem informing the user that this account was deactivated
+			wfe.sendError(response, logEvent, probs.Unauthorized(
+				"An account with the provided public key exists but is deactivated"), nil)
```

Comment:
You also need to add a `return` after `wfe.sendError` or execution will continue after the error is sent over the HTTP response channel.